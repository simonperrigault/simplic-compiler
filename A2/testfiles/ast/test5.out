Program
  List
    FuncDecl
      IdDecl ID="main"
      List
      List
        DeclStmt
          List
            Decl
              IdDecl ID="x"
              Opt
                Numeral NUMERAL="0"
        IfStmt
          EqExpr
            IdUse ID="x"
            Numeral NUMERAL="0"
          List
            AssignStmt
              IdDecl ID="x"
              Numeral NUMERAL="1"
          List
        IfStmt
          LtExpr
            IdUse ID="x"
            Numeral NUMERAL="2"
          List
          List
            FuncCallStmt
              FuncCall
                IdUse ID="print"
                List
                  FuncParam
                    Numeral NUMERAL="2"
