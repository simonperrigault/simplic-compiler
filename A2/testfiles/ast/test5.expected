Program
  List
    FuncDecl
      IdDecl ID="main"
      List
      List
        Decl
          IdDecl ID="x"
          Opt
            Numeral NUMERAL="0"
        IfStmt
          EqExpr
            IdUse ID="x"
            Numeral NUMERAL="0"
          List
            Assignment
              IdDecl ID="x"
              Numeral NUMERAL="1"
          List
        IfStmt
          LtExpr
            IdUse ID="x"
            Numeral NUMERAL="2"
          List
          List
            FuncCallStmt
              FuncCall
                IdUse ID="print"
                List
                  FuncParam
                    Numeral NUMERAL="2"